Assets/_Functionalities/AmsterdamTimeMachine/Scripts/Ui/AtmInformationList.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Collections.Generic;
+using KindMen.Uxios;
+using KindMen.Uxios.Api;
+using UnityEngine;
+using QueryParameters = KindMen.Uxios.Http.QueryParameters;
+
+namespace Netherlands3D.Twin
+{
+    public class AtmInformationList : MonoBehaviour
+    {
+        public class AtmInfo
+        {
+            public string url;
+            public string title;
+            public string startDate;
+            public string endDate;
+            public string textDate;
+            public string thumbnail;
+
+            public override string ToString()
+            {
+                return $"{{url: {url}, title: {title}, startDate: {startDate}, endDate: {endDate}, textDate: {textDate}, thumbnail: {thumbnail}}}";
+            }
+        }
+
+        [SerializeField] private GameObject contentArea;
+        [SerializeField] private AtmInformationListItem photoListItem;
+        [SerializeField] private AtmInformationListItem regularListItem;
+        private const string bagUriTemplate = "https://api.lod.uba.uva.nl/queries/ATM/images-building-nl3d/2/run?bag=http://bag.basisregistraties.overheid.nl/bag/id/pand/{id}";
+        private const string adamLinkTemplate = "https://api.lod.uba.uva.nl/queries/ATM/images-historical-address-nl3d/2/run?address={id}";
+
+        public void LoadFromBagId(string bagId)
+        {
+            QueryParameters parameters = new() {{"id", bagId}};
+
+            PopulateList(new TemplatedUri(bagUriTemplate, parameters));
+        }
+
+        public void LoadFromFeature(FeatureMapping featureMapping)
+        {
+            var adamLinkId = featureMapping.Feature.Properties["id"]?.ToString();
+
+            // check for an id field, otherwise it is not an Amsterdam Time Machine feature
+            if (string.IsNullOrEmpty(adamLinkId)) return;
+            
+            // check if it is a valid Amsterdam Time Machine feature, otherwise we ignore it 
+            if (adamLinkId.StartsWith("https://adamlink.nl/geo/address/") == false) return;
+
+            QueryParameters parameters = new() {{"id", adamLinkId}};
+
+            PopulateList(new TemplatedUri(adamLinkTemplate, parameters));
+        }
+
+        private void PopulateList(Uri url)
+        {
+            contentArea.transform.ClearAllChildren();
+
+            var promise = Resource<List<AtmInfo>>.At(url).Value;
+            promise.Then(CreateListItems);
```

Comment:
puur alleen een vraag, moet die promise niet gechained zijn?

Reply:
Technisch gesproken is deze gechained; de Then methode doet een `return this` waardoor je een fluent interface krijgt (https://en.wikipedia.org/wiki/Fluent_interface) met self-referential returns.

In dit geval herhaal ik de term `promise` omdat Tom heeft aangegeven dat leesbaarder te vinden, en het wat mij betreft een kleine moeite is. Maar functioneel gezien is de werking identiek

Reply:
Ik heb mij niet over ingelezen in de RSG Promises library dat ik exact weet wanneer de Then callbacks getriggered worden; maar ik zou het binnen dezelfde thread proberen te houden. Maar dat is denk ik uberhaupt goed advies als je een proces klaarzet om afgevuurd te worden, de uitvoering mag in aparte threads, maar het klaarzetten doe je in dezelfde / main thread

Reply:
dit is waarom de google libraries chainen omdat alles async op losse threads word gedaan